libs/sdk-ui-dashboard/src/model/queryServices/queryAttributeByDisplayForm.ts

```diff
@@ -0,0 +1,86 @@
+// (C) 2022 GoodData Corporation
+
+import {
+    areObjRefsEqual,
+    IAttributeMetadataObject,
+    ICatalogAttribute,
+    ObjRef,
+    serializeObjRef,
+} from "@gooddata/sdk-model";
+import { QueryAttributeByDisplayForm } from "../queries/attributes";
+import { createCachedQueryService } from "../store/_infra/queryService";
+import { DashboardContext } from "../types/commonTypes";
+import { SagaIterator } from "redux-saga";
+import { call, SagaReturnType, select } from "redux-saga/effects";
+import { invalidQueryArguments } from "../events/general";
+import { selectCatalogAttributes } from "../store";
+
+export const QueryAttributeByDisplayFormService = createCachedQueryService(
+    "GDC.DASH/QUERY.DISPLAY.FORM.ATTRIBUTE",
+    queryService,
+    (query: QueryAttributeByDisplayForm) => {
+        const {
+            payload: { displayForms },
+        } = query;
+
+        return displayForms.map((df) => serializeObjRef(df)).join();
+    },
+);
+
+/**
+ * Loads the attribute metadata for given display form. Primarily the metadata are loaded
+ * from the catalog attributes. If the required attribute is not listed in the catalog
+ * (e.g. deprecated attributes), the attribute metadata are fetched from the backend.
+ */
+async function loadAttributeByDisplayForm(
+    ctx: DashboardContext,
+    catalogAttributes: ICatalogAttribute[],
+    displayForm: ObjRef,
+): Promise<IAttributeMetadataObject> {
+    const { backend, workspace } = ctx;
+    const attribute = catalogAttributes.find((catalogAttribute) =>
+        catalogAttribute.displayForms.some((df) => areObjRefsEqual(df, displayForm)),
+    );
+
+    if (attribute) {
+        return attribute.attribute;
+    }
+
+    return await backend.workspace(workspace).attributes().getAttributeByDisplayForm(displayForm);
```

Comment:
Nitpick: Unless it is in a try catch block, you do not need `return await x`, just `return x` is sufficient.
The next function has also this redundancy.